### PR TITLE
Search for package groups (HMS-4094)

### PR DESCRIPTION
**Pull request by croissanne**
https://github.com/osbuild/image-builder-frontend/assets/11140201/8001d398-6b96-4ee7-bc74-ada93cd5c3d4

Maybe we should do expanded rows, but as a minimal thing let's do this first?

### CODE REVIEW

File: src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx [hunk 1]
```diff
@@ -760,91 +952,202 @@ const Packages = () => {
...
+                rowIndex: rowIndex,
+                onSelect: (event, isSelecting) =>
+                  handleGroupSelect(grp, rowIndex, isSelecting),
+              }}
+            />
+            <Td>
+              @{grp.name}
+              <Popover
+                bodyContent={
+                  <div style={{ height: '40em', overflow: 'scroll' }}>
```
Comment (regexowl) [inline src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx:983]: Let's add a minimum width for the popover to solve this
image

```suggestion
              <Popover
                minWidth="25rem"
                bodyContent={
                  <div style={{ height: '40em', overflow: 'scroll' }}>
```

File: src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx [hunk 2]
```diff
@@ -760,91 +952,202 @@ const Packages = () => {
...
+                    <Table variant="compact">
+                      <Thead>
+                        <Tr>
+                          <Th>Included packages</Th>
```
Comment (regexowl) [inline src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx:987]: Found a group without included packages - is that expected? Would it make sense to add an explanation why is the list empty?

image

File: src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx [hunk 3]
```diff
@@ -760,91 +952,202 @@ const Packages = () => {
...
+              </>
+            ) : grp.repository === 'custom' ? (
+              <>
+                <Td>Third party repository</Td>
+                <Td>Not supported</Td>
+              </>
             ) : (
-              <span className="not-available">Not available</span>
+              <>
+                <Td className="not-available">Not available</Td>
```
Comment (regexowl) [inline src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx:1041]: "Recommended" groups seem to be defaulting to "Not available" for repo and support

image

Should those default to "EPEL" and not supported as it's the only recommendation option currently?

File: src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx [hunk 4]
```diff
@@ -760,91 +952,202 @@ const Packages = () => {
...
+                bodyContent={
+                  <div style={{ height: '40em', overflow: 'scroll' }}>
+                    <Table variant="compact">
+                      <Thead>
+                        <Tr>
+                          <Th>Included packages</Th>
+                        </Tr>
+                      </Thead>
+                      <Tbody>
+                        {grp.package_list.map((pkg) => (
```
Comment (regexowl) [inline src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx:991]: Also maybe we could make "Included packages" a heading of the popover and make only the table body scrollable, but that's very nit picky and I'm happy to open a follow up with suggestion later.

File: src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import React, { useEffect, useMemo, useState } from 'react';
+import React, { ReactElement, useEffect, useMemo, useState } from 'react';
```
Comment (regexowl) [inline src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx:1]: Could we add groups to the Custom repositories modal? The first column could be Packages/groups and then just a name of the group.

image

File: src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import React, { useEffect, useMemo, useState } from 'react';
```
Comment (regexowl) [inline src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx:1]: When a group is selected from recommended and then deselected again the EPEL repo stays selected. It should get deselected when the last item from it is unchecked as for packages.

Also the repo then gets stuck in the list until refresh

unselectable-repo

File: src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx [hunk 7]
```diff
@@ -590,39 +724,68 @@ const Packages = () => {
...
-    }
-    // check for packages that start with the search term
-    if (
-      aPkg.startsWith(debouncedSearchTerm) &&
-      !bPkg.startsWith(debouncedSearchTerm)
-    ) {
-      return -1;
+  ]).sort((a, b) => sortfn(a.name, b.name));
+
+  const transformedGroups = useMemo(() => {
```
Comment (regexowl) [inline src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx:729]: Something probably doesn't get "cleaned" properly. When searching for a package and a group the results will start to mix

mixed-results

**Review by regexowl — COMMENTED** [review]
Nice! Clicked through and added some comments.

**Comment by regexowl** [comment]
I'm perfectly fine with this format as the first step as well btw, we can switch to expandable rows when it's needed

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (5.0.5) is older than required version (6.0.0)

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (5.0.5) is older than required version (6.0.0)

File: src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx [hunk 8]
```diff
@@ -664,6 +848,35 @@ const Packages = () => {
...
+        setIsSelectingGroup(grp);
+      } else {
+        dispatch(addGroup(grp));
+      }
+    } else {
+      dispatch(removeGroup(grp.name));
+      if (
+        isSuccessEpelRepo &&
+        epelRepo.data &&
+        groups.filter((grp) => grp.repository === 'recommended').length === 1
```
Comment (regexowl) [inline src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx:873]: When removing last package group from EPEL, it removes the EPEL repo even when there are still packages added from it.

repo-removed

File: src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx (hunk 6 above)
Comment (regexowl) [inline src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx:1]: EPEL checkbox should be checked and disabled on the Repositories step when some of its package groups is selected on the Packages step as it works for packages now

image

File: src/Components/CreateImageWizardV2/steps/Review/ReviewStepTables.tsx [hunk 9]
```diff
@@ -220,6 +223,19 @@ export const PackagesTable = () => {
                 </Td>
               </Tr>
             ))}
+            {groups.map((grp, grpIndex) => (
+              <Tr key={grpIndex}>
+                <Td className="pf-m-width-30">@{grp.name}</Td>
+                <Td>{grp.description ? grp.description : 'Not available'}</Td>
+                <Td className="pf-m-width-30">
+                  {grp.repository === 'distro'
```
Comment (regexowl) [inline src/Components/CreateImageWizardV2/steps/Review/ReviewStepTables.tsx:231]: Recommended package groups have "Not available" in the repo column

image

**Review by regexowl — COMMENTED** [review]
Found three more things, but other than those I'd say looks great :sparkles:

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (5.0.5) is older than required version (6.0.0)

**Review by regexowl — APPROVED** [review]
Looks great, thank you!

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (5.0.5) is older than required version (6.0.0)

**Pull request merged**